### ISSUE TITLE
test: Use Synapse with Simplified Sliding Sync in our integration tests

### ISSUE THREAD

**Issue by Hywan** [matrix-org/matrix-rust-sdk#3698]
In https://github.com/matrix-org/matrix-rust-sdk/pull/3676, we are migrating the SDK to use Simplified Sliding Sync instead of Sliding Sync itself.

When the PR was written, the version of Synapse used in our integration tests didn't have Simplified Sliding Sync, but was still using the Sliding Sync Proxy. As rightly suggested by @bnjbvr in https://github.com/matrix-org/matrix-rust-sdk/pull/3676#discussion_r1677576900, we should not forget to move away from the Proxy once everything is ready inside Synapse.

## Step 1

Run tests with Synapse instead of the sliding sync proxy:

* [x] https://github.com/matrix-org/matrix-rust-sdk/pull/3983

## Step 2

We now need to migrate the following tests (that are failing since the migration to Synapse):

* [x] `sliding_sync::room::test_room_avatar_group_conversation`
  * [x] https://github.com/matrix-org/matrix-rust-sdk/pull/4032
* [x] `sliding_sync::room::test_room_notification_count`
  * [x] https://github.com/ruma/ruma/pull/1914
  * [x] https://github.com/matrix-org/matrix-rust-sdk/pull/4031
* [ ] `sliding_sync::room::test_delayed_decryption_latest_event`
* [ ] `sliding_sync::room::test_room_info_notable_update_deduplication`
* [x] `sliding_sync::room::test_room_preview`
  * [x] https://github.com/matrix-org/matrix-rust-sdk/pull/4032
* [x] `sliding_sync::notification_client::test_notification`
  * [x] Blocked by https://github.com/element-hq/synapse/issues/17754
  * [x] The fix https://github.com/matrix-org/matrix-rust-sdk/pull/4029
  * [x] https://github.com/ruma/ruma/pull/1913

Some tests seem to be fixed with Synapse 1.115:

* [x] https://github.com/matrix-org/matrix-rust-sdk/pull/4030

---

* Address https://github.com/matrix-org/matrix-rust-sdk/issues/3647

**Comment by poljar** [comment]
I believe this was closed by https://github.com/matrix-org/matrix-rust-sdk/pull/3983.

**Comment by Hywan** [comment]
Not really. `test_delayed_decryption_latest_event` and `test_room_info_notable_update_deduplication` are still marked as `#[ignore]` 🫤.
